apps/nextjs-website/src/components/organisms/FragmentProvider/FragmentProvider.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {
+  ReactNode,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+type FragmentContextType = {
+  fragment: string | null;
+  setFragment: React.Dispatch<React.SetStateAction<string | null>>;
+};
+
+export const FragmentContext = React.createContext<FragmentContextType | null>(
+  null
+);
+
+type FragmentProviderProps = {
+  children: ReactNode;
+};
+
+export function FragmentProvider({ children }: FragmentProviderProps) {
+  const [fragment, setFragment] = useState<string | null>(() =>
+    typeof window !== 'undefined' ? window.location.hash : null
+  );
+
+  const fragmentChangeHandler = useCallback(() => {
+    setFragment(window.location.hash);
+  }, []);
+
+  useEffect(() => {
+    window.addEventListener('hashchange', fragmentChangeHandler);
+    return () => {
+      window.removeEventListener('hashchange', fragmentChangeHandler);
+    };
```

Comment:
Does this function need the return? Or it can be simplified as follows?

``` Typescript
  useEffect(() => window.addEventListener('hashchange', fragmentChangeHandler));
```